Hazel/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -172,5 +210,66 @@ namespace Hazel {
 	{
 		return m_Data.VSync;
 	}
+	
+	void WindowsWindow::SetWindowMode(const WindowMode& mode)
+	{
+		HZ_CORE_ASSERT(m_Window, "Failed to retrieve window.");
+		if (mode == m_Data.Mode) return;
+
+		GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();
+		GLFWvidmode baseVideoMode = *(glfwGetVideoMode(primaryMonitor));
+
+		int windowPosX = 0, windowPosY = 0;
+		GLFWmonitor* monitor = nullptr;
+		int windowDecorated = GLFW_TRUE;
+		int windowResizeable = GLFW_TRUE;
+		int windowAutoIconify = GLFW_FALSE;
+
+		switch (mode)
+		{
+			default:
+				HZ_CORE_ERROR("WindowMode not supported.");
+			case WindowMode::Windowed:
+			{
+				m_Data.Width = m_Data.WindowedWidth;
+				m_Data.Height = m_Data.WindowedHeight;
+				windowPosX = m_Data.WindowedPos.x;
+				windowPosY = m_Data.WindowedPos.y;
+				break;
+			}
+			case WindowMode::Borderless:
+			{
+				m_Data.Width = baseVideoMode.width;
+				m_Data.Height = baseVideoMode.height;
+				monitor = primaryMonitor;
+				windowDecorated = GLFW_FALSE;
+				windowResizeable = GLFW_FALSE;
+				windowAutoIconify = GLFW_TRUE;
+				break;
+			}
+			case WindowMode::FullScreen:
+			{
+				// Achieved by creating a windowed window with size of monitor
+				if (m_Data.Mode == WindowMode::Borderless)
+					// TODO: temorary disable resize events
+					SetWindowMode(WindowMode::Windowed); // this way we obtain a window again
+
+				glfwMaximizeWindow(m_Window);
+				// TODO:
+				//	- Extract width/height removed from taskbar as well (could be 0 if it is hidden)
+				//	- Accomodate this in WindowPosCallback (compensate for fullscreen) not changing position
```

Comment:
In the [`glfwSetWindowPosCallback`](https://github.com/TheCherno/Hazel/pull/108/files#diff-cf3004152bb2411e416be89c25ac3db2R99-R106) we are only checking for position (0,0). This is only the case in borderless mode. In fullscreen we have to accomodate for:

- When the task bar is at the top/left and not hidden, this will offset the position
- We have decorations on top (the window header), this will offset the y-position.

Reply:
This results in a bug:
Go from borderless mode to windowed mode results in the coorect dimensions, but it will render it at position (0,0) instead of correct position where it was located before.

Reply:
This seems to only happen when going from FullScreen -> Borderless -> Windowed. I think this is because going from FullScreen to Borderless doesn't call `glfwRestoreWindow`. This also causes the next FullScreen mode switch to not maximize since the window is still considered maximized.